### PR TITLE
Sprint automatically create meetings

### PR DESCRIPTION
**Pull request by DavidKang**
When we create a new sprint, we also generate the corresponding meetings for this sprint

### CODE REVIEW

File: app/models/sprint.rb [hunk 1]
```diff
@@ -1,7 +1,27 @@
...
+  private
+
+  def no_weekend_day
+    return unless weekend?(start_date.try(:cwday)) || weekend?(end_date.try(:cwday))
+
+    errors[:base] << 'The dates of the sprint could not start/end on weekend'
+  end
+
+  def weekend?(date)
+    [0, 6, 7].include?(date)
```
Comment (bgeuken) [inline app/models/sprint.rb]: It's just some syntactical sugar, but ``` date.in?([0, 6, 7]) ``` might be nicer here. At least to me;-)

File: app/services/create_meetings.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  def self.run(sprint)
+    create_plannings(sprint.start_date)
+    create_standups(sprint.start_date + 1, sprint.end_date - 1)
+    create_grooming(sprint.start_date, sprint.end_date - 1)
+    create_review(sprint.end_date)
+    create_retrospective(sprint.end_date)
+  end
+
+  def self.create_standups(start_date, end_date)
+    (start_date..end_date).each do |date|
```
Comment (bgeuken) [inline app/services/create_meetings.rb]: If you use ``` ... ``` you don't need to reduce the end value

Comment (bgeuken) [reply app/services/create_meetings.rb]: I would also add a comment here, that on the first and last day of our sprint the standups would collide with our planning / review meeting. Which is why we skip the standup on these days.

File: spec/services/create_meetings_spec.rb [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  let(:start_date) { Date.parse('2017-08-07') }
+  let(:end_date) { Date.parse('2017-08-18') }
+
+  context '#self.run' do
+    context 'normal sprint of 2 weeks' do
+      let(:sprint) { create(:sprint) }
+
+      before do
+        sprint
+      end
```
Comment (bgeuken) [inline spec/services/create_meetings_spec.rb]: let!(:sprint) ;-)

File: spec/services/create_meetings_spec.rb [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  end
+
+  context '#self.create_standups' do
+    before do
+      CreateMeetings.create_standups(start_date, end_date)
+    end
+
+    subject { Meeting.where(event_type: :standup) }
+
+    it { expect(subject.count).to eq(10) }
```
Comment (bgeuken) [inline spec/services/create_meetings_spec.rb]: Matter of taste. I would prefer to write this in one line.

Comment (bgeuken) [reply spec/services/create_meetings_spec.rb]: Shouldn't it be 8 for a 10 day sprint? First and last day doesn't have a standup

Comment (DavidKang) [reply spec/services/create_meetings_spec.rb]: right!

**Comment by DavidKang** [comment]
@bgeuken, please re-review it :smile_cat:

**Pull request merged**